test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/ClassInclusionReport.java

```diff
@@ -37,7 +37,10 @@ public static ClassInclusionReport load() {
         TreeSet<String> set = new TreeSet<>();
         try (Scanner scanner = new Scanner(usedClassesReport.toFile())) {
             while (scanner.hasNextLine()) {
-                set.add(scanner.nextLine());
+                // Starting with GraalVM for JDK 24 the format of the report has changed prefixing each line with
+                // the class loader name and a colon. We need to strip that part.
+                String[] line = scanner.nextLine().split(":");
+                set.add(line[line.length - 1]);
```

Comment:
Given the comment, I would rather use `indexOf()` and `substring()`? Or you are absolutely sure there won't be any other colon?

Reply:
For the time being it's at most one colon, something like `[<classloader>:]<classname>`, that's why I went for the simplest approach. If more colons show up in the future it will still work as long as the classname still appears last in the sequence (and the classname can't contain a colon). If something gets added after the classname then the test will break and we will notice.

I don't think it's worth "validating" the file's format, at least not for the time being.